src/Covenant/Internal/PrimType.hs

```diff
@@ -223,18 +221,19 @@ typeOfThreeArgFunc =
     VerifyEcdsaSecp256k1Signature -> (TyByteString, TyByteString, TyByteString, TyBoolean)
     VerifySchnorrSecp256k1Signature -> (TyByteString, TyByteString, TyByteString, TyBoolean)
     IfThenElse -> (TyBoolean, TyPlutusData, TyPlutusData, TyPlutusData)
```

Comment:
Is 'TyPlutusData' really right here? I'm not sure why the builtin ifte should be restricted to Plutus Data. Maybe it just has to be this way to make the (Haskell) types line up?

Reply:
This is a holdover from the old system: I've only left these in place to avoid breaking stuff. The new system in `Covenant.Type` and `Covenant.Prim` should replace these.